instance/models/load_balancer.py

```diff
@@ -111,6 +145,55 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return self.domain
 
+    def modifies_configuration(method):  # nopep8 # pylint: disable=no-self-argument
+        """
+        Flag an internal method as one that modifies the state of load balancer configuration,
+        to allow managing configuration state outside and around the method call.
+        """
+        @functools.wraps(method)
+        def wrapper(self, *args, **kwargs):
+            """
+            Used to wrap the function that modified load balancer configuration state.
+            """
+            try:
+                # Prevent race conditions by updating state only if the LB is actually clean or dirty.
+                updated = LoadBalancingServer.objects.filter(
+                    pk=self.pk,
+                    configuration_state__in=[self.CLEAN, self.DIRTY]
+                ).update(
+                    configuration_state=self.IN_PROGRESS
+                )
+
+                # A race condition or just a normal queuing up of LB reconfiguration requests is handled.
```

Comment:
Note by 'queueing up' I'm referring to a bulk queue -- the periodic task will of course "handle" these "queued up" requests in a batch through a single reconfiguration.